packages/opentelemetry-plugin-document-load/src/documentLoad.ts

```diff
@@ -0,0 +1,283 @@
+
+/**
+ * This class represents a document load plugin
+ */
+export class DocumentLoad extends BasePlugin<unknown> {
+  readonly component: string = 'document-load';
+  readonly version: string = '1';
+  moduleName = this.component;
+  protected _config!: PluginConfig;
+
+  /**
+   *
+   * @param {DocumentLoadConfig} config
+   */
+  constructor(config: PluginConfig = {}) {
+    super();
+    this._onDocumentLoaded = this._onDocumentLoaded.bind(this);
+    this._config = config;
+  }
+
+  /**
+   * callback to be executed when page is loaded
+   * @private
+   */
+  private _onDocumentLoaded() {
+    this._collectPerformance();
+  }
+
+  /**
+   * Collects information about performance and creates appropriate spans
+   * @private
+   */
+  private _collectPerformance() {
+    const entries = this._getEntries();
+
+    const rootSpan = this._startSpan(
+      AttributeNames.DOCUMENT_LOAD,
+      PerformanceTimingNames.FETCH_START,
+      entries
+    );
+
+    this._startAndFinishSpan(
+      AttributeNames.DOMAIN_LOOKUP,
+      PerformanceTimingNames.DOMAIN_LOOKUP_START,
+      PerformanceTimingNames.DOMAIN_LOOKUP_END,
+      entries,
+      {
+        parent: rootSpan,
+      }
+    );
+
+    // Opening Connection
+    const connectSpan = this._startSpan(
+      AttributeNames.CONNECT,
+      PerformanceTimingNames.CONNECT_START,
+      entries,
+      {
+        parent: rootSpan,
+      }
+    );
+
+    // TLS negotiation
+    if (hasKey(entries, PerformanceTimingNames.SECURE_CONNECTION_START)) {
+      const value = entries[PerformanceTimingNames.SECURE_CONNECTION_START];
+      if (typeof value === 'number' && value > 0) {
+        this._startAndFinishSpan(
+          AttributeNames.CONNECT_SECURE,
+          PerformanceTimingNames.SECURE_CONNECTION_START,
+          PerformanceTimingNames.CONNECT_END,
+          entries,
+          {
+            parent: connectSpan,
+          }
+        );
+      }
+    }
+    // Connection negotiation ends
+    this._endSpan(connectSpan, PerformanceTimingNames.CONNECT_END, entries);
+
+    // Cache seek plus response time
+    this._startAndFinishSpan(
+      AttributeNames.CACHE_SEEK,
+      PerformanceTimingNames.FETCH_START,
+      PerformanceTimingNames.RESPONSE_END,
+      entries,
+      {
+        parent: rootSpan,
+      }
+    );
+
+    // TTFB - time to first byte
+    this._startAndFinishSpan(
+      AttributeNames.TTFB,
+      PerformanceTimingNames.REQUEST_START,
+      PerformanceTimingNames.RESPONSE_START,
+      entries,
+      {
+        parent: rootSpan,
+      }
+    );
+
+    // Response time only (download)
+    this._startAndFinishSpan(
+      AttributeNames.RESPONSE_TIME,
+      PerformanceTimingNames.RESPONSE_START,
+      PerformanceTimingNames.RESPONSE_END,
+      entries,
+      {
+        parent: rootSpan,
+      }
+    );
+
+    this._endSpan(rootSpan, PerformanceTimingNames.LOAD_EVENT_START, entries);
```

Comment:
Would you be willing to create issues to track additional timings including:
- resource timings (CSS, images loads, etc.)
- time to first paint, time to first contentful paint, DOM content loaded, DOM interactive
?